UNITY/trailRendererRandom/script/trailRendererRandom.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class trailRendererRandom : MonoBehaviour
+{
+	[SerializeField]private TrailRenderer trailRenderer;
+	Gradient gradient;
+	GradientColorKey[] colorKey;
+	GradientAlphaKey[] alphaKey;
+
+	private Start () {
+		if(trailRenderer == null) {
```

Comment:
Add this header [RequireComponent(typeof(TrailRenderer))] at the top of your class, to be sure that you have always TrailRenderer on your gameobject, so you just have to GetComponent<TrailRendere> in case it's missing